version: '3'

services:
  db:
    image: kartoza/postgis:11.5-2.5
    container_name: db_{{project_name}}
    restart: unless-stopped
    env_file:
      - ./env/db.env
    ports:
      - "5432:5432"
    volumes:
      - "pg-data-{{project_name}}:/var/lib/postgresql"
    networks:
      -cartoview-internal:
  rabbitmq:
    image: rabbitmq:latest
    container_name: rabbitmq_{{project_name}}
    restart: unless-stopped
    ports:
      - "5672:5672"
      - "15672:15672"
    networks:
      -cartoview-internal:
  data-dir-conf:
    image: geonode/geoserver_data:2.16.2
    container_name: geoserver_data_dir_{{project_name}}
    restart: on-failure
    container_name: gsconf4cartoview
    volumes:
      - geoserver-dir-{{project_name}}:/geoserver_data/data
  geoserver:
    image: geonode/geoserver:2.16.2
    container_name: geoserver_{{project_name}}
    depends_on:
      - db
      - data-dir-conf
    restart: unless-stopped
    ports:
      - "8080:8080"
    volumes:
      - geoserver-dir-{{project_name}}:/geoserver_data/data
    env_file:
      - ./env/tomcat.env
    networks:
      -cartoview-external:
        ipv4_address: 10.5.0.2
      -cartoview-internal:
  cartoview:
    build: ./docker/cartoview/
    image: cartoview_{{project_name}}:latest
    container_name: cartoview_{{project_name}}
    command: bash -c "python server.py"
    restart: unless-stopped
    hostname: cartoview
    depends_on:
      - db
      - geoserver
    env_file: ./env/django.env
    volumes:
      - "./:/usr/src/carto_app"
    ports:
      - "8000:8000"
    networks:
      -cartoview-external:
        ipv4_address: 10.5.0.3
      -cartoview-internal:
  celery:
    image: cartoview_{{project_name}}:latest
    container_name: celery_{{project_name}}
    command: bash -c "celery worker --app={{project_name}} --broker=amqp://guest:guest@rabbitmq:5672/ -B -E -l debug"
    restart: unless-stopped
    hostname: celery
    depends_on:
      - db
      - rabbitmq
      - geoserver
    stdin_open: true
    env_file:
      - ./env/django.env
    volumes:
      - "./:/usr/src/carto_app"
    networks:
      -cartoview-internal:
  nginx:
    image: nginx:alpine
    container_name: nginx_{{project_name}}
    restart: unless-stopped
    ports:
      - "80:80"
    volumes:
      - ./conf/nginx/nginx.conf:/etc/nginx/nginx.conf
    networks:
      -cartoview-external:
        ipv4_address: 10.5.0.4
      -cartoview-internal:
networks:
    -cartoview-external:
      driver: bridge
      ipam:
        config:
          - subnet: 10.5.0.0/16
    -cartoview-internal:
       internal: true
volumes:
  geoserver-dir-{{project_name}}:
  pg-data-{{project_name}}:
